geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/SpatialTiledRasterLayer.scala

```diff
@@ -411,20 +411,30 @@ class SpatialTiledRasterLayer(
 
   def toGeoTiffRDD(
     tags: Tags,
-    geotiffOptions: GeoTiffOptions
+    resampleMethod: ResampleMethod,
+    decimations: List[Int],
+    geoTiffOptions: GeoTiffOptions
   ): JavaRDD[Array[Byte]] = {
-    val mapTransform = MapKeyTransform(
-      rdd.metadata.layout.extent,
-      rdd.metadata.layout.layoutCols,
-      rdd.metadata.layout.layoutRows)
+    val mapTransform =
+      MapKeyTransform(
+        rdd.metadata.layout.extent,
+        rdd.metadata.layout.layoutCols,
+        rdd.metadata.layout.layoutRows
+      )
 
     val crs = rdd.metadata.crs
 
-    val geotiffRDD = rdd.map { x =>
-      val transKey = ProjectedExtent(mapTransform(x._1), crs)
+    val geotiffRDD =
+      rdd.map { case (k, v) =>
+        val transKey = ProjectedExtent(mapTransform(k), crs)
 
-      (x._1, MultibandGeoTiff(x._2, transKey.extent, transKey.crs, tags, geotiffOptions).toByteArray)
-    }
+        val geoTiff = {
+          val gtiff: GeoTiff[MultibandTile] = MultibandGeoTiff(v, transKey.extent, transKey.crs, tags, geoTiffOptions)
+          gtiff.withOverviews(resampleMethod, decimations)
```

Comment:
Not really worth a change but it would be my idiomatic and less text just to smash these together.

```scala
MultibandGeoTiff(v, transKey.extent, transKey.crs, tags, geoTiffOptions)
    .withOverviews(resampleMethod, decimations)
```